src/TransferOption.js

```diff
@@ -0,0 +1,82 @@
+import React, { useRef } from 'react'
+import cx from 'classnames'
+import propTypes from '@dhis2/prop-types'
+
+import { colors } from './theme.js'
+
+const DOUBLE_CLICK_MAX_DELAY = 250
+
+export const TransferOption = ({
+    className,
+    dataTest,
+    label,
+    highlighted,
+    onClick,
+    onDoubleClick,
+    value,
+    additionalData,
+}) => {
+    const doubleClickTimeout = useRef(null)
+
+    return (
+        <div
+            data-test={dataTest}
+            onClick={event => {
+                const option = { label, value, ...additionalData }
+
+                if (doubleClickTimeout.current) {
+                    clearTimeout(doubleClickTimeout.current)
+                    doubleClickTimeout.current = null
+
+                    onDoubleClick({ option }, event)
+                } else {
+                    doubleClickTimeout.current = setTimeout(() => {
+                        clearTimeout(doubleClickTimeout.current)
+                        doubleClickTimeout.current = null
+                    }, DOUBLE_CLICK_MAX_DELAY)
```

Comment:
Should we have to take unmounting after a click but before the timeout into account? Or will that not be an issue?

Reply:
I guess the worst they could do is remove the current property, in which case you're passing undefined to clearTimeout and setting current to null.